server/controllers/DataAccessLayer.js

```diff
@@ -102,6 +110,145 @@ exports.UpdateUser = function(user) {
+        cachedReports.push(report);
+    }
+    // return cachedReports
+    return cachedReports;
+}
+
+/**
+ * Adds the given Report to the local text file
+ * @param {Report} report The supplied Report object to be added to the local text file
+ */
+exports.AddReportToFile = function(report) {
+    // Gets the formatted string containing the chat logs
+    let chatLogString = GetChatLogString(report.chatLogs);
+    // Gets a new date for the lastUpdatedDate
+    let newLastUpdatedDate = new Date()
+    // Create a string to store in the text file as a Report
+    let reportString = report.id + "|||" + report.offendingUserId + "|||" + 
+                     report.submittingUserId + "|||" + report.reportType + "|||" + 
+                     report.reportComment + "|||" + report.dateSubmitted.toISOString() + "|||" + 
+                     newLastUpdatedDate.toISOString() + "|||" + report.status + "|||" + chatLogString + "|||\n";
+    // Append the string to the text file
+    fs.appendFileSync('data/Reports.txt', reportString);
+
+    // Add the report to the cache
+    cachedReports.push(report);
+}
+
+/**
+ * Updates the cache and local text file to update to represent all changes made to the Report object passed in
+ * @param {Report} report The Report object that has had values updated, cache and local file will update based on id
+ */
+exports.UpdateReport = function(report) {
+    // Find the index in the cache for the original User object
+    let index = cachedReports.findIndex(x => x.id === report.id);
+    // Get a pointer to the original User object
+    let originalReport = cachedReports[index];
+
+    // Create a regex that will be searched for within the text file
+    let originalReportStringRegex = ".*" + originalReport.id + ".*";
+
+    // Create a new Date object to set the lastUpdatedDate's to
+    let newLastUpdatedDate = new Date();
+
+    let newChatLogString = GetChatLogString(report.chatLogs);
+
+    // Create a new string to update the file to
+    let newReportString = report.id + "|||" + report.offendingUserId + "|||" + 
+                        report.submittingUserId + "|||" + report.reportType + "|||" + 
+                        report.reportComment + "|||" + report.dateSubmitted.toISOString() + "|||" + 
+                        newLastUpdatedDate.toISOString() + "|||" + report.status + "|||" + newChatLogString + "|||";
+
+    // Create a regexp to find the correct contents to change
+    const regex = new RegExp(originalReportStringRegex, "g");
+
+    // Create an options object for the replace function
+    let replaceOptions = {
+        files: "data/Reports.txt",
+        from: regex,
+        to: newReportString
+    }
+
+    // Replace the relevant contents of the text file (replace-in-file node package)
+    replace(replaceOptions)
+        .then(result => {
+            // If no error occurred and the result states that the file was successfully changed
+            if (result.hasChanged === true) {
```

Comment:
Is the `===` check necessary?

Reply:
Yes, it allows for verification that the file was successfully updated with errors, then I update the cache

Reply:
Can't you just do `if (result.hasChanged)`?